core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/controlplane/services/policydefinition/PolicyDefinitionQueryValidatorFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ * Contributors:
+ *       Contributors to the Eclipse Foundation - initial API and implementation
+ */
+
+package org.eclipse.edc.connector.controlplane.services.policydefinition;
+
+import org.eclipse.edc.connector.controlplane.policy.spi.PolicyDefinition;
+import org.eclipse.edc.connector.controlplane.services.query.QueryValidator;
+import org.eclipse.edc.policy.model.AndConstraint;
+import org.eclipse.edc.policy.model.AtomicConstraint;
+import org.eclipse.edc.policy.model.Constraint;
+import org.eclipse.edc.policy.model.Expression;
+import org.eclipse.edc.policy.model.LiteralExpression;
+import org.eclipse.edc.policy.model.MultiplicityConstraint;
+import org.eclipse.edc.policy.model.OrConstraint;
+import org.eclipse.edc.policy.model.XoneConstraint;
+
+import java.util.List;
+import java.util.Map;
+
+public class PolicyDefinitionQueryValidatorFactory {
+    public static QueryValidator createQueryValidator() {
```

Comment:
pls replace the constructor call in `PolicyDefinitionServiceImpl` with this, to make this PR consistent.

Reply:
Please add Javdoc

Reply:
OK. But I would include this in another commit after adding every `QueryValidatorFactory`s instead to make the commit history look clear.

Reply:
Please keep PRs self-contained. The Javadoc should be added in a commit to this PR.

Reply:
> Please keep PRs self-contained. The Javadoc should be added in a commit to this PR.

As for the Javadoc I would squash into one commit before force-push.

Reply:
> > pls replace the constructor call in PolicyDefinitionServiceImpl with this, to make this PR consistent.
> 
> OK. But I would include this in another commit after adding every `QueryValidatorFactory`s instead to make the commit history look clear.

I would prefer to have a PR introducing code that's actually being used, because otherwise it is dead code, and if, for example, you'll stop working on this project next week, that dead code could stay there for a long time.

Reply:
> I would prefer to have a PR introducing code that's actually being used, because otherwise it is dead code, and if, for example, you'll stop working on this project next week, that dead code could stay there for a long time.

Makes sense.

I meant introducing this in another commit within the same PR. Changing construction call before finishing all `QueryValidatorFactory`s makes commit message harder to write and understand.

Reply:
the commit message for this would be something like "move instantiation logic into a separated factory" or something like this, it is totally readable and, above all, consistent.

TBH in this PR I would also inject the query validator in the policy definition service and simplifying the service tests, like the issue #3731 required.
This would be the way to split the issue "vertically" in multiple, independent tasks (one for every service)

Reply:
BTW: please note that we "squash" the PRs, so also if you do 10 commits, after merge there will be just one commit, the message will be the name of the PR (sidenote: the current one must be changed to adhere to our conventions)